src/allomancy/java/leaf/cosmere/allomancy/common/manifestation/AllomancyBrass.java

```diff
@@ -27,14 +30,67 @@ public AllomancyBrass(Metals.MetalType metalType)
 	protected void applyEffectTick(ISpiritweb data)
 	{
 		int mode = getMode(data);
-
 		String uuid = data.getLiving().getStringUUID();
-		if (mode > 0 && !playerThreadMap.containsKey(uuid))
+
+		// data thread management
 		{
-			playerThreadMap.put(uuid, new BrassThread(data));
+			if (mode > 0 && !playerThreadMap.containsKey(uuid))
+			{
+				playerThreadMap.put(uuid, new BrassThread(data));
+			}
+
+			playerThreadMap.entrySet().removeIf(entry -> !entry.getValue().isRunning || AllomancyEntityThread.serverShutdown);
 		}
 
-		playerThreadMap.entrySet().removeIf(entry -> !entry.getValue().isRunning || AllomancyEntityThread.serverShutdown);
+		// data processing
+		{
+			// check if brass is off or compounding
+			if (mode <= 0)
+			{
+				return;
+			}
+
+			// this is the only way to check if the player is still online, thanks forge devs
+			if (data.getLiving().level.getServer().getPlayerList().getPlayer(data.getLiving().getUUID()) == null)
+			{
+				return;
+			}
+
+			//todo, replace x * mode with config based value
+			double allomanticStrength = getStrength(data, false);
+
+			//put on a different tick to zinc
+			boolean isActiveTick = getActiveTick(data) % 2 == 0;
+			if (isActiveTick)
+			{
+				if (playerThreadMap.get(uuid) == null)
+				{
+					playerThreadMap.put(uuid, new BrassThread(data));
+				}
+
+				List<LivingEntity> entitiesToAffect = playerThreadMap.get(uuid).requestEntityList();
+				for (LivingEntity e : entitiesToAffect)
+				{
+					if (e instanceof Mob mob)
+					{
+						mob.setNoAi(mode == 3 && allomanticStrength > 15);
+
+						switch (mode)
```

Comment:
Just realised, this switch case might have to do with #64. There's no breaks, unless that's intended?

Reply:
The original intention was to be able to have it do all of them depending on mode, flaring would still turn off aggressiveness and forget the last attacker.

This still needs changing, because it's obviously not working correctly.

Reply:
Alright, I'll leave it for a future PR